h2/connection.py

```diff
@@ -1387,7 +1387,9 @@ def _receive_headers_frame(self, frame):
             # compatibility, catch all of them.
             raise ProtocolError("Error decoding header block: %s" % e)
 
-        headers = validate_headers(headers)
+        headers = validate_headers(headers,
+                                   client_side=self.client_side,
+                                   is_trailer=False)
```

Comment:
While we're here, this call is probably unnecessary: it's a duplicate of the call in the stream, so you can probably lose it.